### PR TITLE
FE-1245 - Removed the "Authenticate for SPF" section from domain details page and verification page for bounce and sending/bounce

### PR DESCRIPTION
**Pull request by shrirupa**
FE-1245 - Removed the "Authenticate for SPF" section from domain details page and verification page for bounce and sending/bounce

### What Changed
 - Removed the "Authenticate for bounce" section from
      - Unverified domain details page - domains/details/sending-bounce/test.123.900
      - Verification pages 
             - Sending/Bounce = domains/details/test.123.900/verify-sending-bounce
             - Bounce = http://localhost:3100/domains/details/test.123.900/verify-bounce
       - Copy in the left Panel for Bounce is also updated to match - https://sparkpost.invisionapp.com/console/share/CS1SX9XT3H/478945608

### How To Test
 - Verify the Authentication for SPF section is gone, and rest of functionality still works correctly

### To Do
- [ ] Address feedback

### CODE REVIEW

File: src/pages/domains/components/SetupBounceDomainSection.js [hunk 1]
```diff
@@ -17,26 +17,14 @@ const PlaneIcon = styled(Send)`
...
+  const { verify, showAlert, userName, isByoipAccount, verifyBounceLoading } = useDomains();
   const readyFor = resolveReadyFor(status);
   const initVerificationType = isByoipAccount && status.mx_status === 'valid' ? 'MX' : 'CNAME';
   const bounceDomainsConfig = getConfig('bounceDomains');
   const { control, handleSubmit, watch, register } = useForm();
   const watchVerificationType = watch('verificationType', initVerificationType);
-  const [warningBanner, toggleBanner] = useState(true);
 
   const onSubmit = () => {
     if (!readyFor.bounce) {
```
Comment (kchu1337) [inline src/pages/domains/components/SetupBounceDomainSection.js:30]: Is there still a need for this condition? Or is it just for safety?

File: src/pages/domains/components/SetupBounceDomainSection.js [hunk 2]
```diff
@@ -127,7 +94,7 @@ export default function SetupBounceDomainSection({ domain, isSectionVisible, tit
               provider.
               <Panel.Action>
                 <ExternalLink
-                  to={`mailto:?subject=Assistance%20Requested%20Verifying%20a%20Sending%20Domain%20on%20SparkPost&body=${userName}%20has%20requested%20your%20assistance%20verifying%20a%20sending%20domain%20with%20SparkPost.%20Follow%20the%20link%20below%20to%20find%20the%20values%20you%E2%80%99ll%20need%20to%20add%20to%20the%20settings%20of%20your%20DNS%20provider.%0D%0A%5BGo%20to%20SparkPost%5D(${window.location})%0D%0A`}
+                  to={`mailto:?subject=Assistance%20Requested%20Verifying%20a%20Bounce%20Domain%20on%20SparkPost&body=${userName}%20has%20requested%20your%20assistance%20verifying%20a%20bounce%20domain%20with%20SparkPost.%20Follow%20the%20link%20below%20to%20find%20the%20values%20you%E2%80%99ll%20need%20to%20add%20to%20the%20settings%20of%20your%20DNS%20provider.%0D%0A%5BGo%20to%20SparkPost%5D(${window.location})%0D%0A`}
```
Comment (kchu1337) [inline src/pages/domains/components/SetupBounceDomainSection.js:97]: Is this just a typo you are fixing now?

File: src/pages/domains/components/SendingAndBounceDomainSection.js [hunk 3]
```diff
@@ -42,7 +37,6 @@ export default function SendingAndBounceDomainSection({ domain, isSectionVisible
 
   const getButtonText = () => {
     if (!readyFor.dkim || !readyFor.bounce) return 'Authenticate Domain';
-    if (status.spf_status !== 'valid') return 'Authenticate for SPF';
   };
```
Comment (kchu1337) [inline src/pages/domains/components/SendingAndBounceDomainSection.js:40]: Do you still need this? The condition for displaying the button looks the same;  `(!readyFor.bounce || !readyFor.dkim)`

**Review by kchu1337 — CHANGES_REQUESTED** [review]
2 additional things I think you can delete. And 1 question.

File: src/pages/domains/components/SetupBounceDomainSection.js (hunk 2 above)
Comment (shrirupa) [reply src/pages/domains/components/SetupBounceDomainSection.js:97]: yes

File: src/pages/domains/components/SetupBounceDomainSection.js (hunk 1 above)
Comment (shrirupa) [reply src/pages/domains/components/SetupBounceDomainSection.js:30]: Removed it. you are right, submit button is already guarded by this condition.

File: src/pages/domains/components/SendingAndBounceDomainSection.js (hunk 3 above)
Comment (shrirupa) [reply src/pages/domains/components/SendingAndBounceDomainSection.js:40]: Removed it! Thanks!

**Review by kchu1337 — APPROVED** [review]

**Review by kchu1337 — APPROVED** [review]

**Pull request merged**